modules/nw-cudn-cr.adoc

```diff
@@ -0,0 +1,156 @@
++
+[source, yaml]
+----
+apiVersion: k8s.ovn.org/v1
+kind: ClusterUserDefinedNetwork
+metadata:
+  name: <cudn_name> # <1>
+spec:
+  namespaceSelector: # <2>
+    matchLabels: # <3>
+    - "<example_namespace_one>":"" # <4>
+    - "<example_namespace_two>":"" # <4>
+  network: # <5>
+    topology: Layer2 # <6>
+    layer2: # <7>
+      role: Primary # <8>
+      subnets:
+        - "2001:db8::/64"
+        - "10.100.0.0/16" # <9>
+----
+<1> Name of your `ClusterUserDefinedNetwork` resource. This should not be `default` or a duplicate of any resource in your cluster.
+<2> A label query over the set of namespaces that the cluster UDN applies to. Uses the standard Kubernetes `MatchLabel` selector. Must not point to `default` or `openshift-*` namespaces.
+<3> Uses the `matchLabels` selector type, where terms are evaluated with an _*AND*_ relationship. 
+<4> Because the `matchLabels` selector type is used, provisions namespaces matching both `<example_namespace_one>` _and_ `<example_namespace_two>`.
+<5> Describes the network configuration.
+<6> The `topology` field describes the network configuration; accepted values are `Layer2` and `Layer3`. Specifying a `Layer2` topology type creates one logical switch that is shared by all nodes.
+<7> This field specifies the topology configuration. It can be `layer2` or `layer3`.
+<8> Specifies `Primary` or `Secondary`. `Primary` is the only `role` specification supported in {product-version}.
+<9> For `Layer2` topology types the following specifies config details for the `subnet` field:
++
+* The subnets field is optional.
+* The subnets field is of type `string` and accepts standard CIDR formats for both IPv4 and IPv6.
+* The subnets field accepts one or two items. For two items, they must be of a different family. For example, subnets values of `10.100.0.0/16` and `2001:db8::/64`.
+* `Layer2` subnets can be omitted. If omitted, users must configure IP addresses for the pods. As a consequence, port security only prevents MAC spoofing.
+* The `Layer2` `subnets` field is mandatory when the `ipamLifecycle` field is specified.
++
+.. Create a YAML file, such as `cluster-layer-three-udn.yaml`, to define your request for a `Layer3` topology as in the following example:
++
+[source, yaml]
+----
+apiVersion: k8s.ovn.org/v1
+kind: ClusterUserDefinedNetwork
+metadata:
+  name: <cudn_name> # <1>
+spec:
+  namespaceSelector: # <2>
+    matchExpressions: # <3>
+    - key: kubernetes.io/metadata.name # <4>
+      operator: In # <5>
+      values: ["<example_namespace_one>, <example_namespace_two>"] # <6>
+  network: # <7>
+    topology: Layer3 # <8>
+    layer3: # <9>
+      role: Primary # <10>
+      subnets: # <11>
+        - cidr: 10.100.0.0/16
+          hostSubnet: 64
+----
+<1> Name of your `UserDefinedNetwork` resource. This should not be `default` or duplicate any global namespaces created by the Cluster Network Operator (CNO).
```

Comment:
> Name of your `UserDefinedNetwork` resource.

ClusterUserDefienedNetwork*

> This should not be `default` 

Same as mentioned above, I am not aware of such restrictions on metadata.name.

> or duplicate any global namespaces created by the Cluster Network Operator (CNO).

I dont undestand what it means or any limitations on the naming related to CNO.

@tssurya do you aware of such restrictions?

Reply:
nope no idea @ormergi don't know any restrictions

Reply:
Updated to just <1> Name of your `ClusterUserDefinedNetwork` resource.